src/gigs-board/components/posts/PostEditor.jsx

```diff
@@ -186,18 +187,68 @@ const normalizeLabel = (label) =>
     .toLowerCase()
     .trim("-");
 
+const checkLabel = (label) => {
+  Near.asyncView(nearDevGovGigsContractAccountId, "is_allowed_to_use_labels", {
+    editor: context.accountId,
+    labels: [label],
+  }).then((allowed) => {
+    if (allowed) {
+      State.update({ warning: "" });
+    } else {
+      State.update({
+        warning:
+          'The label "' +
+          label +
+          '" is protected and can only be added by moderators',
+      });
+      return;
+    }
+  });
+};
+
 const setLabels = (labels) => {
   labels = labels.map((o) => {
     o.name = normalizeLabel(o.name);
     return o;
   });
-  let labelStrings = labels.map((o) => {
-    return o.name;
-  });
-  State.update({ labels, labelStrings });
+  if (labels.length < state.labels.length) {
+    let oldLabels = new Set(state.labels.map((label) => label.name));
+    for (let label of labels) {
+      oldLabels.delete(label.name);
+    }
+    let removed = oldLabels.values().next().value;
+    Near.asyncView(
+      nearDevGovGigsContractAccountId,
+      "is_allowed_to_use_labels",
+      { editor: context.accountId, labels: [removed] }
+    ).then((allowed) => {
+      if (allowed) {
+        let labelStrings = labels.map((o) => {
+          return o.name;
+        });
```

Comment:
```suggestion
        let labelStrings = labels.map(({ name }) => name);
```

Apply the same above and below.

P.S. What is the reason to call the variable `o`?